operator/src/test/java/org/bf2/operator/managers/ImagePullSecretManagerTest.java

```diff
@@ -120,4 +124,66 @@ void testWithSecrets() {
                         .collect(Collectors.toSet()));
     }
 
+    @Test
+    void testPullSecretTypeChangePropagatedAsNewSecret() {
+        String ns = "testPullSecretTypeChangePropagatedAsNewSecret";
+        Secret originalNamespacedSecret = client.secrets()
+            .inNamespace(ns)
+            .create(new SecretBuilder()
+                    .withNewMetadata()
+                        .withName("mk-pull-name")
+                    .endMetadata()
+                    .withType("kubernetes.io/dockercfg")
+                    .withData(Map.of("key", "value"))
+                    .build());
+
+        Secret replacement = new SecretBuilder()
+            .withNewMetadata()
+                .withName("name")
+            .endMetadata()
+            .withType("kubernetes.io/dockerconfigjson")
+            .withData(Map.of("key", "value"))
+            .build();
+
+        ManagedKafka managedKafka = new ManagedKafka();
+        managedKafka.setMetadata(new ObjectMetaBuilder().withName("mk").withNamespace(ns).build());
+
+        imagePullSecretManager.propagateSecrets(client, managedKafka, List.of(replacement));
+
+        Secret replacementNamespacedSecret = client.secrets().inNamespace(ns).withName("mk-pull-name").get();
+
+        assertNotEquals(originalNamespacedSecret.getMetadata().getUid(), replacementNamespacedSecret.getMetadata().getUid());
+    }
+
+    @Test
+    void testPullSecretTypeUnchangedUpdatesExistingSecret() {
+        String ns = "testPullSecretTypeUnchangedUpdatesExistingSecret";
+
+        Secret originalNamespacedSecret = client.secrets()
+            .inNamespace(ns)
+            .create(new SecretBuilder()
+                    .withNewMetadata()
+                        .withName("mk-pull-name")
+                    .endMetadata()
+                    .withType("kubernetes.io/dockercfg")
+                    .withData(Map.of("key", "value"))
+                    .build());
+
+        Secret replacement = new SecretBuilder()
+            .withNewMetadata()
+                .withName("name")
+            .endMetadata()
+            .withType("kubernetes.io/dockercfg")
+            .withData(Map.of("key", "value"))
+            .build();
+
+        ManagedKafka managedKafka = new ManagedKafka();
+        managedKafka.setMetadata(new ObjectMetaBuilder().withName("mk").withNamespace(ns).build());
+
+        imagePullSecretManager.propagateSecrets(client, managedKafka, List.of(replacement));
+
+        Secret replacementNamespacedSecret = client.secrets().inNamespace(ns).withName("mk-pull-name").get();
+
+        assertEquals(originalNamespacedSecret.getMetadata().getUid(), replacementNamespacedSecret.getMetadata().getUid());
```

Comment:
I think asserting the `uid` is unchanged is necessary but not sufficient we also want to assert that the key or value change was propagated. 

As things stand we could pass this test with a no-op implementation.

Reply:
Added some new assertions with the last commit.